src/txn/txn_timestamp.c

```diff
@@ -879,6 +879,49 @@ __wt_txn_set_read_timestamp(WT_SESSION_IMPL *session, wt_timestamp_t read_ts)
     return (0);
 }
 
+/*
+ * __txn_set_timestamp_internal --
+ *     Does the "heavy lifting" for the outside-facing txn_set_timestamp functions.
+ */
+static int
+__txn_set_timestamp_internal(
+  WT_SESSION_IMPL *session, uint64_t commit, uint64_t durable, uint64_t prepare, uint64_t read)
+{
+    WT_CONNECTION_IMPL *conn;
+    bool set_ts;
+
+    conn = S2C(session);
+    set_ts =
+      commit != WT_TS_NONE || durable != WT_TS_NONE || prepare != WT_TS_NONE || read != WT_TS_NONE;
+
+    /* Look for a commit timestamp. */
+    if (commit != WT_TS_NONE)
+        WT_RET(__wt_txn_set_commit_timestamp(session, commit));
+
+    /*
+     * Look for a durable timestamp. Durable timestamp should be set only after setting the commit
+     * timestamp.
+     */
+    if (durable != WT_TS_NONE)
+        WT_RET(__wt_txn_set_durable_timestamp(session, durable));
+    __wt_txn_publish_durable_timestamp(session);
```

Comment:
This code will publish a durable timestamp even if none were set via the API call. Is that necessary and deliberate?

Reply:
It appears as though the old code did that too, but it seems surprising, so maybe we should consider changing the behavior now if it's safe?

Reply:
I think this is covered in the above thread?

Reply:
I think it's safe to publish all the time, but it's odd to publish when there isn't a chance it will be updated.

The name of the publish function is also kind of misleading - I took a closer look and it will publish either a commit or durable timestamp, so this code needs some cleanup (either here, or in another ticket).